### ISSUE TITLE
TFuncKey type breaks when used in Record and Map types

### ISSUE THREAD

**Issue by trevordunn** [i18next/i18next#1944]
## 🐛 Bug Report

Use `TFuncKey` in a `Record` or `Map` type, and the outputted type from `t(theKey)` will not be `string`.

## To Reproduce

https://codesandbox.io/p/sandbox/elastic-burnell-wm183x?file=%2Fsrc%2FApp.tsx

Notice that there are TS errors at the bottom of `<Component>` and `<App>`, and that `t(recordKey)` and `t(mapKey)` equal `'string | { bar: string; }'`.

Curiously, `t(key)` also errors out when used in a `function Foo() {}` component instead of a `const Foo = () => {}` component. Not sure what the difference is there.

## Expected behavior

The TS type from `t(recordKey)` and `t(mapKey)` should equal `string`.

## Your Environment
- "i18next": "^22.4.13"
- "react": "^18.2.0"
- "react-dom": "^18.2.0"
- "react-i18next": "^12.2.0"
- "@types/i18n-js": "^3.8.2"
- "@types/react": "^18.0.15"
- "@types/react-dom": "^18.0.6"
- "typescript": "^4.6.4"

**Comment by jamuhl** [comment]
Could be you're running into this: https://www.i18next.com/overview/typescript#argument-of-type-defaulttfuncreturn-is-not-assignable-to-parameter-of-type-xyz (as `t` function can return null the type does not just return a string as you like)

**Comment by trevordunn** [comment]
@jamuhl Thanks for the idea, but unfortunately no that's not it. As you can tell by running the example that all strings are being outputted. You can also hover over all the vars and confirm that none of the types have `null` in them. To triple check I did add in the `returnNull: false,` configuration and I'm still seeing the problem.

**Comment by trevordunn** [comment]
This is still an issue.

**Comment by trevordunn** [comment]
This is still an issue.

**Comment by trevordunn** [comment]
This is still an issue.

**Comment by adrai** [comment]
this will probably still be an "issue" with: https://github.com/i18next/i18next/pull/1911

**Comment by adrai** [comment]
Please try with i18next v23.0.1 and react-i18next v13.0.0

**Comment by trevordunn** [comment]
@adrai Do you know what we're supposed to use for `TFuncKey` now? I'm getting this error:
>Module '"i18next"' has no exported member 'TFuncKey'.

**Comment by adrai** [comment]
@trevordunn https://www.i18next.com/misc/migration-guide
Try with `ParseKeys`
<img width="401" alt="image" src="https://github.com/i18next/i18next/assets/1086194/1766bd5e-9dda-40c0-b493-2b6003c37f37">

**Comment by trevordunn** [comment]
@adrai It doesn't look like `ParseKeys` is a drop-in replacement for `TFuncKey`. I'm trying to replicate what `TFuncKey` was doing before, but I'm not sure if I've got it right. Does this look right to you? Seems to work, though I am confused a bit about what I'm supposed to put in for the 2nd and 3rd params.
`type TFuncKey = ParseKeys<'translation', {}, 1>;`

**Comment by adrai** [comment]
first is the namespace, second the toptions, and third the keyprefix... also used here: https://github.com/i18next/react-i18next/blob/master/icu.macro.d.ts#L10
@pedrodurek maybe you can tell more about it?

**Comment by pedrodurek** [comment]
That's correct! We could also assign default values in another PR, so you could omit the argument.

**Comment by adrai** [comment]
@pedrodurek just done so: https://github.com/i18next/i18next/commit/18ce83f7f5a28548cef107c5cad5434ef5033a60

**Comment by trevordunn** [comment]
k I believe this issue has been resolved with i18next v23.0.1 and react-i18next v13.0.0.

**Comment by adrai** [comment]
> @pedrodurek just done so: [18ce83f](https://github.com/i18next/i18next/commit/18ce83f7f5a28548cef107c5cad5434ef5033a60)

fyi: released in v23.1.0
